examples/tv-app/linux/include/audio-output/AudioOutputManager.cpp

```diff
@@ -22,36 +22,75 @@ using namespace std;
 using namespace chip::app;
 using namespace chip::app::Clusters::AudioOutput;
 
+AudioOutputManager::AudioOutputManager()
+{
+    mCurrentOutput = 1;
+
+    for (int i = 1; i < 4; ++i)
+    {
+        OutputInfoType outputInfo;
+        outputInfo.outputType = chip::app::Clusters::AudioOutput::OutputTypeEnum::kHdmi;
+        outputInfo.name       = chip::CharSpan::fromCharString("HDMI");
+        outputInfo.index      = static_cast<uint8_t>(i);
+        mOutputs.push_back(outputInfo);
+    }
+}
+
 uint8_t AudioOutputManager::HandleGetCurrentOutput()
 {
-    return 0;
+    return mCurrentOutput;
 }
 
 CHIP_ERROR AudioOutputManager::HandleGetOutputList(AttributeValueEncoder & aEncoder)
 {
     // TODO: Insert code here
-    return aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
-        int maximumVectorSize = 3;
-        for (int i = 0; i < maximumVectorSize; ++i)
+    std::vector<OutputInfoType> outputs = mOutputs;
+    return aEncoder.EncodeList([outputs](const auto & encoder) -> CHIP_ERROR {
+        for (auto const & outputInfo : outputs)
         {
-            chip::app::Clusters::AudioOutput::Structs::OutputInfo::Type outputInfo;
-            outputInfo.outputType = chip::app::Clusters::AudioOutput::OutputTypeEnum::kHdmi;
-            outputInfo.name       = chip::CharSpan::fromCharString("exampleName");
-            outputInfo.index      = static_cast<uint8_t>(1 + i);
             ReturnErrorOnFailure(encoder.Encode(outputInfo));
         }
         return CHIP_NO_ERROR;
     });
 }
 
+bool isAudioOutputIndexInRange(const uint8_t index, std::vector<OutputInfoType> outputs)
```

Comment:
Should this really pass the vector by value?

Reply:
Updated.